app/stac_api/collection_temporal_extent.py

```diff
@@ -114,22 +120,41 @@ def update_start_temporal_extent_on_item_update(
             # set earliest start_datetime to min(earliest_start_datetime
             # of all items but the one currently updated and
             # new_start_datetime).
-            if bool(qs.filter(properties_start_datetime__isnull=False)):
+            logger.warning(
+                'Looping over all items of collection %s,'
+                'to update temporal extent, this may take a while',
+                collection.name
+            )
+            qs_other_items_with_properties_start_datetime = qs.filter(
+                properties_start_datetime__isnull=False
+            )
+            if bool(qs_other_items_with_properties_start_datetime):
```

Comment:
Uses `exists()` instead of bool(). Because if it exists we reconstruct a new query, using exists might have a performance impact here.

Reply:
`bool()` evaluates a qs, so that when the qs is accessed the next time, the cache is used.
`exist()` does not evaluate the qs. So I used `bool()` instead of `exist()` in all those places, where the qs is accessed after calling `bool()`. In other cases I used `exist()`

Reply:
I don't know the detail of the implementation of the queryset, but if we would later loop over the qs, then bool() would be for sure correct. However when we use the qs to do a new qs, which is here the case with the `.earliest()`, django might improve performance with using another query and we don't need to evaluate the complete initial query.

Reply:
My understanding is, that `.earliest()` accesses the already existing qs. If using `.filter()` then I agree, `exists()` would be better here. But I thought, using `.earliest()` is just a kind of sorting and returning the first hit and not creating a new qs.
Since this in an important point I'll have to double check here. This would impact a lot of `exists()` and `bools()` of this PR.

Reply:
Mmm based on the docu, I don't think that `earliest()` and `latest()` evaluates the whole queryset like `bool()` does: https://docs.djangoproject.com/en/3.1/ref/models/querysets/#when-querysets-are-evaluated

Reply:
This is what I thought initially as well.
But checking the docu for `filter()` it is explicitly stated, that a new QuerySet is returned:
https://docs.djangoproject.com/en/3.1/ref/models/querysets/#filter
--> "Returns a new QuerySet containing objects..."

Which is not true for the docu on `earliest()` or `latest()` respectively:
https://docs.djangoproject.com/en/3.1/ref/models/querysets/#django.db.models.query.QuerySet.latest
--> "Returns the latest object in the table..."*

This is why I thought using `bool()` first, which evaluates the qs, followed by an `earliest()` on this already evaluated qs is more performant than using `exists()` instead of `bool()` in this cases.
In all other cases I sticked to using `exists()`. I'll have to dig deeper here, but I did not find any more useful docu so far.

Reply:
Maybe this would be better : 
```python
try:
    earliest = qs.filter(properties_start_datetime__isnull=False).earliest('properties_start_datetime')
except DoesNotExists as error:
   ...
```
Not 100% sure if this impact performance, but here we only retrieve one object, while with `bool(qs.filter(properties_start_datetime__isnull=False))` you potentially evaluates a big queryset. @boecklic  what is your opinion based on your experience with django ?

Reply:
Funny, this is pretty much what I first had in mind, too.
But again, I guess the first part `qs.filter(properties_start_datetime__isnull=False)` is the expensive part here, which will be evaluated anyway and this is the part where there can be many many hits. And afterwards the `earliest()` simply sorts and returns one of the many. So I guess this will not significantly improve performance.

Reply:
Perhaps it might make sense to create a test db with a very large number of sample items, so I can check the costs of these queries..

Reply:
OK, so now there is  a ticket for possible improvements/refactoring of `collection_temporal.extent.py` (https://jira.swisstopo.ch/browse/BGDIINF_SB-1582).

As for the point of using `bool()` or `exists()` here, I did some test on a collection with 150'000 items:

**exists()**
```
qs_exists = Item.objects.filter(collection_id="3").exclude(id="0")
qs_exists.exists()
```
this yields: 
```(0.001) SELECT (1) AS "a" FROM "stac_api_item" WHERE ("stac_api_item"."collection_id" = 3 AND NOT ("stac_api_item"."id" = 0)) LIMIT 1; args=(3, 0)
```
fetching the earliest datetime with `qs_exists.earliest('properties_datetime').properties_datetime` yields:
```
(0.170) SELECT "stac_api_item"."id", "stac_api_item"."name", "stac_api_item"."collection_id", "stac_api_item"."geometry"::bytea, "stac_api_item"."created", "stac_api_item"."updated", "stac_api_item"."properties_datetime", "stac_api_item"."properties_start_datetime", "stac_api_item"."properties_end_datetime", "stac_api_item"."properties_title", "stac_api_item"."etag", "stac_api_collection"."id", "stac_api_collection"."name", "stac_api_collection"."created", "stac_api_collection"."updated", "stac_api_collection"."description", "stac_api_collection"."extent_geometry"::bytea, "stac_api_collection"."extent_start_datetime", "stac_api_collection"."extent_end_datetime", "stac_api_collection"."license", "stac_api_collection"."summaries", "stac_api_collection"."title", "stac_api_collection"."etag" FROM "stac_api_item" INNER JOIN "stac_api_collection" ON ("stac_api_item"."collection_id" = "stac_api_collection"."id") WHERE ("stac_api_item"."collection_id" = 3 AND NOT ("stac_api_item"."id" = 0)) ORDER BY "stac_api_item"."properties_datetime" ASC LIMIT 1; args=(3, 0)
True
```
During this fetch with `earliest()` the qs is evaluated and the time needed is 0.170 ms.
After evaluation, the cached qs is used. Average time per fetching the `earliest()` after initial evaluation is 0.124 (average of 10 fetches).


**bool()**
```
qs_bool = Item.objects.filter(collection_id="3").exclude(id="0")
bool(qs_bool)
```
yields:
(0.931) SELECT "stac_api_item"."id", "stac_api_item"."name", "stac_api_item"."collection_id", "stac_api_item"."geometry"::bytea, "stac_api_item"."created", "stac_api_item"."updated", "stac_api_item"."properties_datetime", "stac_api_item"."properties_start_datetime", "stac_api_item"."properties_end_datetime", "stac_api_item"."properties_title", "stac_api_item"."etag", "stac_api_collection"."id", "stac_api_collection"."name", "stac_api_collection"."created", "stac_api_collection"."updated", "stac_api_collection"."description", "stac_api_collection"."extent_geometry"::bytea, "stac_api_collection"."extent_start_datetime", "stac_api_collection"."extent_end_datetime", "stac_api_collection"."license", "stac_api_collection"."summaries", "stac_api_collection"."title", "stac_api_collection"."etag" FROM "stac_api_item" INNER JOIN "stac_api_collection" ON ("stac_api_item"."collection_id" = "stac_api_collection"."id") WHERE ("stac_api_item"."collection_id" = 3 AND NOT ("stac_api_item"."id" = 0)); args=(3, 0)
True

The check for existence takes 0.931 ms, as the qs is fully evaluated at the same time, compared to 0.170ms when using `exists()` (where no evaluation is done).

Fetching earliest datetime using `earliest()` on the already evaluated qs with qs_bool.earliest('properties_datetime').properties_datetime` takes 0.118 ms.
Average of 10 subsequent executions: 0.13 ms.

**Executive summary ;-)**
So this is purely empirical guess-work, but `exist()` seems to be a bit faster in this setting, as the initial evaluation is quicker in comparison to `bool()`. So I'll use `exist()` here. And when we run into performance or other issues in future, I guess we'll do the refactoring and probably have to think about asynchronous jobs then.